gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -1860,13 +1870,21 @@ gnc_plugin_page_register_set_filter (GncPluginPage *plugin_page, const gchar *fi
     GNCLedgerDisplayType ledger_type;
     GNCLedgerDisplay *ld;
     Account *leader;
-    gchar *default_filter = g_strdup_printf("%s,%s,%s,%s", DEFAULT_FILTER, "0", "0", "0");
+    gchar *default_filter;
+    gchar *default_num_str;
 
     priv = GNC_PLUGIN_PAGE_REGISTER_GET_PRIVATE(plugin_page);
     ld = priv->ledger;
     ledger_type = gnc_ledger_display_type (ld);
     leader = gnc_ledger_display_leader (ld);
 
+    if (ledger_type == LD_GL)
+        default_num_str = "30";
+    else
+        default_num_str = "0";
+
+    default_filter = g_strdup_printf("%s,%s,%s,%s", DEFAULT_FILTER, "0", "0", default_num_str);
+
```

Comment:
Thanks for the update. It works as advertised now.

Your most recent change on the other hand now introduces a maintenance issue: it uses the same two "magic" number in 3 different locations (30 and 0).

Please factor this test out in a separate internal function that takes the ledger type as input and returns the filter_default_days as a const char*.

That will allow you to create default_filter directly (as it was) like
```const gchar *default_num_str = ledger_default_filter_days (ledger_type);
gchar *default_filter = g_strdup_printf("%s,%s,%s,%s", DEFAULT_FILTER, "0", "0", default_num_str);```

You can use #define's or const char *'s at the top of the source to setup sensible names for the two magic numbers themselves.

Reply:
Have made the changes, did not see that at all.